src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -0,0 +1,1181 @@
+    }
+} // end of class Base
+");
+
+            verifier.VerifyTypeIL("Derived", @"
+.class private auto ansi beforefieldinit Derived
+    extends Base
+{
+    .custom instance void System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+        01 00 00 00
+    )
+    // Fields
+    .field private int32 '<Prop>k__BackingField'
+    .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+        01 00 00 00
+    )
+    // Methods
+    .method public hidebysig specialname virtual 
+        instance int32 get_Prop () cil managed 
+    {
+        .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+            01 00 00 00
+        )
+        // Method begins at RVA 0x2069
+        // Code size 7 (0x7)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: ldfld int32 Derived::'<Prop>k__BackingField'
+        IL_0006: ret
+    } // end of method Derived::get_Prop
+    .method public hidebysig specialname virtual 
+        instance void set_Prop (
+            int32 'value'
+        ) cil managed 
+    {
+        .custom instance void [netstandard]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+            01 00 00 00
+        )
+        // Method begins at RVA 0x2071
+        // Code size 8 (0x8)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: ldarg.1
+        IL_0002: stfld int32 Derived::'<Prop>k__BackingField'
+        IL_0007: ret
+    } // end of method Derived::set_Prop
+    .method public hidebysig specialname rtspecialname 
+        instance void .ctor () cil managed 
+    {
+        // Method begins at RVA 0x207a
+        // Code size 7 (0x7)
+        .maxstack 8
+        IL_0000: ldarg.0
+        IL_0001: call instance void Base::.ctor()
+        IL_0006: ret
+    } // end of method Derived::.ctor
+    // Properties
+    .property instance int32 Prop()
+    {
+        .custom instance void System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 Derived::get_Prop()
+        .set instance void Derived::set_Prop(int32)
+    }
+} // end of class Derived
+");
+        }
+    }
+
+    [Fact]
+    public void RequiredMemberAttributeEmitted_AddRequiredOnOverride()
+    {
+        var comp = CreateCompilationWithRequiredMembers(@"
+class Base
+{
+    public virtual int Prop { get; set; }
+}
+class Derived : Base
+{
+    public override required int Prop { get; set; }
+}
+class DerivedDerived : Derived
+{
+    public override required int Prop { get; set; }
+}
+");
+
+        var expectedRequiredMembers = new[] { "Derived.Prop", "DerivedDerived.Prop" };
+
+        var verifier = CompileAndVerify(comp, sourceSymbolValidator: ValidateRequiredMembersInModule(expectedRequiredMembers));
+        verifier.VerifyDiagnostics();
+
+
+        if (ExecutionConditionUtil.IsCoreClr)
```

Comment:
Using `VerifyTypeIL` and only doing so based on `IsCoreClr` seems unusual. Why not do specific metadata checks about where the RequiredMember attribute was emitted, as we typically do?

Reply:
> Using `VerifyTypeIL` and only doing so based on `IsCoreClr` seems unusual. Why not do specific metadata checks about where the RequiredMember attribute was emitted, as we typically do?

I only do so on framework because whether types come from `[netstandard]` or `[msorlib]` changes, which caused the test failures on the first few iterations of this PR. I do do attribute checks as well, through `ValidateRequiredMembersInModule`, and when the next PR adds support for decoding these from metadata I'll add these checks to look at metadata symbols as well, not just source.

Reply:
I understand that the output of `VerifyTypeIL` depends on the platform. I think a better solution is to avoid `VerifyTypeIL` (except for special tests) in favor of our usual validation helpers.